### PR TITLE
Fix misc. typos

### PR DESCRIPTION
**Pull request by luzpaz**
Found via `codespell -q 3 -I ../sympy-whitelist.txt`where wiltelist contains:
```
ang
ans
cas
dum
german
iff
ith
lightyear
lightyears
nd
numer
numers
ot
sinc
```

NO ENTRY

### CODE REVIEW

File: doc/src/tutorial/intro.rst [hunk 1]
```diff
@@ -111,7 +111,7 @@ matrices, and much, much more, and do it all symbolically.  It includes
 modules for plotting, printing (like 2D pretty printed output of math
 formulas, or `\LaTeX`), code generation, physics, statistics, combinatorics,
 number theory, geometry, logic, and more. Here is a small sampling of the sort
-of symbolic power SymPy is capable of, to whet your appetite.
```
Comment (smichr) [inline doc/src/tutorial/intro.rst]: This was correct: whet/sharpen appetite vs wet your whistle.

File: sympy/calculus/finite_diff.py [hunk 2]
```diff
@@ -148,7 +148,7 @@ def finite_diff_weights(order, x_list, x0=S.One):
     derivative is wanted, weights for 0th, 1st and 2nd order are
     calculated "for free", so are formulae using subsets of ``x_list``.
     This is something one can take advantage of to save computational cost.
-    Be aware that one should define ``x_list`` from nearest to farest from
+    Be aware that one should define ``x_list`` from nearest to farthest from
```
Comment (smichr) [inline sympy/calculus/finite_diff.py]:
```suggestion
    Be aware that one should define ``x_list`` from nearest to furthest from
```

Comment (smichr) [reply sympy/calculus/finite_diff.py]: furthest/adj
farthest/adv

**Comment by smichr** [comment]
A couple of comments and then you will have to edit your commit message to remove the RELEASE NOTES entry there. I added it to the OP (which is where it should go).

File: doc/src/tutorial/intro.rst (hunk 1 above)
Comment (luzpaz) [reply doc/src/tutorial/intro.rst]: TIL, thanks. Will revert this

**Comment by smichr** [comment]
could you grep for `factization`, too? It appears at least once in sparse.py

**Comment by czgdp1807** [comment]
@luzpaz Can you please re-word the commit message for https://github.com/sympy/sympy/commit/36f9b5336df291f707504711861f86a299bef1a9. Refer https://github.com/sympy/sympy/pull/17158#issuecomment-508942964 too.
For re-wording use, refer, https://coderwall.com/p/_thg9a/reword-a-git-commit-message.

**Comment by luzpaz** [comment]
> factization
 https://github.com/sympy/sympy/blob/ef94e5588332f1d8d1aded63423456a50e323cdc/sympy/matrices/sparse.py#L551  

What should it be replaced with?

**Comment by smichr** [comment]
factization` -> `factorization`...but there is work to be done in matrices (#17156) so let's save a build and do it later.

**Pull request merged**